### PR TITLE
Attempt to ensure params are const and not modified

### PR DESCRIPTION
**Pull request by ZeldaZach**
WIP PR

### CODE REVIEW

File: mtgjson4/mtg_builder.py [hunk 1]
```diff
@@ -263,7 +265,9 @@ def determine_layout_and_div_name(soup: bs4.BeautifulSoup,
         if c_variations:
             card_info['variations'] = c_variations
 
-    async def build_legalities_part(self, card_mid: int, card_info: mtg_global.CardDescription) -> None:
+        return card_info
+
+    async def build_legalities_part(self, card_mid: int) -> None:
```
Comment (Daenyth) [inline mtgjson4/mtg_builder.py]: Need to adjust the return type here.

File: mtgjson4/mtg_builder.py [hunk 2]
```diff
@@ -333,6 +348,14 @@ def determine_layout_and_div_name(soup: bs4.BeautifulSoup,
         if c_original_text:
             card_info['originalText'] = c_original_text
 
+        return card_info
+
+    @staticmethod
+    def combine_all_parts_into_a_card(main_parts, original_parts, legal_parts, foreign_parts) -> Dict[str, Any]:
+        dict_builder1 = {**main_parts, **original_parts}
+        dict_builder2 = {**legal_parts, **foreign_parts}
+        return {**dict_builder1, **dict_builder2}
```
Comment (Daenyth) [inline mtgjson4/mtg_builder.py]: why not `return {**main_parts, **original_parts, **legal_parts, **foreign_parts}`?

You should also `-> CardDescription` on this method so you can push the return type up the stack from there

File: tox.ini [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = yapf-check,mypy,lint,isort-check,unit
+envlist = yapf-inplace,mypy,lint
```
Comment (Daenyth) [inline tox.ini:2]: no `unit` when you run `tox`?

**Review by Daenyth — APPROVED** [review]
Looks like a step in the right direction

File: mtgjson4/mtg_builder.py (hunk 2 above)
Comment (ZeldaZach) [reply mtgjson4/mtg_builder.py]: Was based on SO and didn't know if I could do that, thanks for the tip

File: tox.ini (hunk 3 above)
Comment (ZeldaZach) [reply tox.ini:2]: Just for now, didn't see unit doing anything

**Pull request merged**